ml-agents-envs/mlagents/envs/sampler_class.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np 
+from abc import ABC, abstractmethod
+
+from .exception import SamplerException
+
+class Sampler(ABC): 
+
+    @abstractmethod
+    def sample_parameter(self, *args, **kwargs):
+        pass
+
+
+class UniformSampler(Sampler):
+    def __init__(self, param_dict):
```

Comment:
Make the min and max value explicit arguments to `__init__`, then do `sampler_cls(**param_dict)` in the factory instead.

Reply:
I think all of the samplers should either take a random seed, or an instance of a random.Random (or the numpy equivalent). That way you at least have the option of replaying a particular sequence of samples.

Reply:
Hmm, I guess we set np.random's seed at the start. So you can ignore the comment about the seed if you want, and just stuck with numpy.

Reply:
So I tried setting np.random's seed, but there isn't a deterministic selection for the reset parameter sampling. I am guessing this is happening because in sample_all you can call the different sampling methods in a different order in each loop, which changes subsequent values. 

One alternative could be to maintain a seed in each of the sampler methods (i.e. - maintain the seed in each instance of UniformSampler or other sampling classes); what are your thoughts?